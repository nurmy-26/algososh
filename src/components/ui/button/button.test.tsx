import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";


describe('Button', () => {

  test('Отрисовка кнопки без текста', () => {
    render(<Button />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('Отрисовка кнопки с текстом', () => {
    render(<Button text="Развернуть" />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('Отрисовка заблокированной кнопки', () => {
    render(<Button disabled />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toBeDisabled();
    expect(btn).toMatchSnapshot();
  });

  test('Отрисовка кнопки с индикатором загрузки', () => {
    render(<Button isLoader />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('При клике на кнопку вызывается колбэк', () => {
    const handleClick = jest.fn();

    render(<Button onClick={handleClick} text="Рассчитать" />);
    fireEvent.click(screen.getByText(/рассчитать/i));
    expect(handleClick).toBeCalledTimes(1);
  });

})