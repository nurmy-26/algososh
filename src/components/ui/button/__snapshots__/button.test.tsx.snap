// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Отрисовка заблокированной кнопки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Отрисовка кнопки без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Отрисовка кнопки с индикатором загрузки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Отрисовка кнопки с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Развернуть
  </p>
</button>
`;
